### PR TITLE
Fix issue  #25361 select default value of payment term and method and banck  account 

### LINKED ISSUE #25361: Default customer payment terms not taken when creating an invoice
### Bug

The default payment terms, method and Banck Account values set for the customer are not taken when I create an invoice and select the customer from the dropdown list.

### Environment Version

17.0.2

### Environment OS

Apache

### Environment Web server

_No response_

### Environment PHP

8.1.21

### Environment Database

MySQL or MariaDB 10.5.19-MariaDB-0+deb11u2

### Environment URL(s)

/compta/facture/card.php?action=create&leftmenu=

### Expected and actual behavior

Payment Terms, Payment Method and Bank Account should be set in the form according to the values specified in the Customer card.

### Steps to reproduce the behavior

Create a customer and set the Payment Terms, Payment Method and Bank Account
Click on New Invoce within the menu "Billing | Payment menu/Customer invoices/"
Select a third party from the drop down menu.

### Attached files

Default values are taken if I comment the lines 3221 and 3222 of /compta/facture/card.php and add the lines of code taken from a previous version of dolibarr as follows:

					// jQuery("#formtocreateaction").val("create");
					// jQuery("#formtocreate").submit();
					var socid = $(this).val();
			                var fac_rec = $(\'#fac_rec\').val();
        			        window.location.href = "'.$_SERVER["PHP_SELF"].'?action=create&socid="+socid+"&fac_rec="+fac_rec;

### PR DESCRIPTION
**Pull request by lamrani002**
# FIX|Fix #25361
select default value of payment term and method and bank  account when create new invoice and selecting a client who has been already those info declared

### CODE REVIEW

File: htdocs/compta/facture/card.php [hunk 1]
```diff
@@ -3126,7 +3126,7 @@
 
 	// when bank account is empty (means not override by payment mode form a other object, like third-party), try to use default value
 	if ($socid > 0 && $fk_account) {	// A company has already been set and it has a default fk_account
-		$fk_account = GETPOSTISSET('fk_account') ? GETPOST("fk_account", 'int') : $fk_account;	// The GETPOST is used only if form was posted to avoid to take default value, because in such case, the default must be the one of the company
```
Comment (Hystepik) [inline htdocs/compta/facture/card.php:3129]: Here you verify if GETPOST retrun something with GETPOSTISSET and if it returns somthing you you give to $fk_account the value fk_account without verifying

A better fix could be 
$fk_account = !empty($fk_account) ? $fk_account : GETPOST("fk_account", 'int');

File: htdocs/compta/facture/card.php [hunk 2]
```diff
@@ -3682,7 +3682,7 @@ function setRadioForTypeOfIncoice() {
 	// Payment term
 	print '<tr><td class="nowrap fieldrequired">'.$langs->trans('PaymentConditionsShort').'</td><td colspan="2">';
 	print img_picto('', 'payment', 'class="pictofixedwidth"');
-	print $form->getSelectConditionsPaiements((GETPOSTISSET('cond_reglement_id') && GETPOST('cond_reglement_id', 'int') != 0) ? GETPOST('cond_reglement_id', 'int') : $cond_reglement_id, 'cond_reglement_id', -1, 1);
```
Comment (Hystepik) [inline htdocs/compta/facture/card.php:3685]: same as last comment

File: htdocs/compta/facture/card.php [hunk 3]
```diff
@@ -3739,7 +3739,7 @@ function setRadioForTypeOfIncoice() {
 	// Payment mode
 	print '<tr><td>'.$langs->trans('PaymentMode').'</td><td colspan="2">';
 	print img_picto('', 'bank', 'class="pictofixedwidth"');
-	print $form->select_types_paiements((GETPOSTISSET('mode_reglement_id') && GETPOST('mode_reglement_id') != 0)? GETPOST('mode_reglement_id') : $mode_reglement_id, 'mode_reglement_id', 'CRDT', 0, 1, 0, 0, 1, 'maxwidth200 widthcentpercentminusx', 1);
```
Comment (Hystepik) [inline htdocs/compta/facture/card.php:3742]: same as last comment

File: htdocs/compta/facture/card.php [hunk 4]
```diff
@@ -3688,7 +3688,7 @@ function setRadioForTypeOfIncoice() {
 	// Payment term
 	print '<tr><td class="nowrap fieldrequired">'.$langs->trans('PaymentConditionsShort').'</td><td colspan="2">';
 	print img_picto('', 'payment', 'class="pictofixedwidth"');
-	print $form->getSelectConditionsPaiements((GETPOSTISSET('cond_reglement_id') && GETPOST('cond_reglement_id', 'int') != 0) ? GETPOST('cond_reglement_id', 'int') : $cond_reglement_id, 'cond_reglement_id', -1, 1, 0, 'maxwidth500 widthcentpercentminusx');
+	print $form->getSelectConditionsPaiements((!empty('cond_reglement_id') && GETPOST('cond_reglement_id', 'int') != 0) ? GETPOST('cond_reglement_id', 'int') : $cond_reglement_id, 'cond_reglement_id', -1, 1, 0, 'maxwidth500 widthcentpercentminusx');
```
Comment (eldy) [inline htdocs/compta/facture/card.php:3691]: Check on supplier invoice to make the same.

**Comment by lamrani002** [comment]
i had a probleme with pull so i close this PR and make a new one

**Comment by eldy** [comment]
Fixed with https://github.com/Dolibarr/dolibarr/pull/26305

**Pull request closed without merging**